model/src/main/proto/oppia_logger.proto

```diff
@@ -9,90 +9,97 @@ option java_multiple_files = true;
-      HintContext access_hint_context = 10;
+      HintContext access_hint_context = 13;
 
       // The event being logged is related to offering a solution when it becomes available.
-      ExplorationDetailsContext solution_offered_context = 11;
+      ExplorationContext solution_offered_context = 14;
 
       // The event being logged is related to accessing a solution.
-      ExplorationDetailsContext access_solution_context = 12;
+      ExplorationContext access_solution_context = 15;
 
       // The event being logged is related to submitting an answer.
-      SubmitAnswerContext submit_answer_context = 13;
+      SubmitAnswerContext submit_answer_context = 16;
 
       // The event being logged is related to playing a voiceover.
-      PlayVoiceOverContext play_voice_over_context = 14;
+      PlayVoiceOverContext play_voice_over_context = 17;
 
       // The event being logged is related to backgrounding of the application.
-      LearnerDetailsContext app_in_background_context = 15;
+      LearnerDetailsContext app_in_background_context = 18;
 
       // The event being logged is related to foregrounding of the application.
-      LearnerDetailsContext app_in_foreground_context = 16;
+      LearnerDetailsContext app_in_foreground_context = 19;
 
       // The event being logged is related to exiting an exploration.
-      ExplorationDetailsContext exit_exploration_context = 17;
+      ExplorationContext exit_exploration_context = 20;
 
       // The event being logged is related to finishing an exploration.
-      ExplorationDetailsContext finish_exploration_context = 18;
+      ExplorationContext finish_exploration_context = 21;
 
       // The event being logged is related to resuming an exploration.
-      LearnerDetailsContext resume_exploration_context = 19;
+      LearnerDetailsContext resume_exploration_context = 22;
 
       // The event being logged is related to restarting an exploration.
-      LearnerDetailsContext start_over_exploration_context = 20;
+      LearnerDetailsContext start_over_exploration_context = 23;
 
       // The event being logged is related to deleting a profile.
-      LearnerDetailsContext delete_profile_context = 21;
+      LearnerDetailsContext delete_profile_context = 24;
+
+      // The event being logged is related to the opening of home activity.
+      NullContext open_home = 25;
+
+      // The event being logged is related to the opening of profile chooser activity.
+      NullContext open_profile_chooser = 26;
     }
   }
 
-  // Structure of an exploration context.
-  message ExplorationContext {
-    // The active topic ID when the event is logged.
-    string topic_id = 1;
-    // The active story ID when the event is logged.
-    string story_id = 2;
-    // The active exploration ID when the event is logged.
-    string exploration_id = 3;
-  }
+  // Representation of a null context which can be used to represent events which don't require
+  // any additional data to be recorded.
```

Comment:
Alternatively, you could use a boolean and document that the value doesn't matter, only its presence (it's a bit of a hack, but we've used it elsewhere and I think I got it from a recommendation outside our team).

I generally use the value 'true' since it makes the code nicer ('true' helps communicate that the specific mode is "enabled").

Reply:
nice one, done -- thanks!